driver-core/src/main/com/mongodb/internal/inject/EmptyProvider.java

```diff
@@ -13,19 +13,29 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+package com.mongodb.internal.inject;
 
-package com.mongodb.internal.connection;
+import com.mongodb.annotations.Immutable;
+
+import java.util.Optional;
+
+@Immutable
+public final class EmptyProvider<T> implements OptionalProvider<T> {
```

Comment:
`ChangeListener.java` was removed, `EmptyProvider.java` was added. This classes have 0 lines in common and are located in different directories, and yet GitHub / Git considers this a change + rename :)